api/v1alpha1/ingressgateway_types.go

```diff
@@ -193,9 +247,10 @@ func (in *IngressGateway) ToConsul(datacenter string) capi.ConfigEntry {
 	return &capi.IngressGatewayConfigEntry{
 		Kind:      in.ConsulKind(),
 		Name:      in.ConsulName(),
-		TLS:       in.Spec.TLS.toConsul(),
+		TLS:       *in.Spec.TLS.toConsul(),
```

Comment:
Do we need to worry about nil pointer dereferences here?

Reply:
this file was copy-pasted over, so i think we should be in the clear?